### ISSUE TITLE
Pretify status report

### ISSUE THREAD

**Issue by laraSalconiDenis** [go-gitea/gitea#28189]
### Feature Description

Not sure if it's intended but currently the commit statuses does not render markdown.  
My current usage of gitea is:
- gitea 1.20.3
- jenkins 2.422 + gitea_checks 449.v0672c320ee47

the two picture link before and after, most statuses are fill up and send automatically by the CICD an the plugin.  
![before](https://github.com/go-gitea/gitea/assets/38858375/5551ea6a-425f-4e78-acd6-9371871f5d0c)

# Solution
to fix the render I had to copy the `templates/repo/pulls/status.tmpl` and change the line [L40](https://github.com/go-gitea/gitea/blob/1075ff74b5050f671c5f9824ae39390230b3c85d/templates/repo/pulls/status.tmpl#L40). 
```
{{RenderMarkdownToHtml $.Context .Description}}
```
## Result
![after](https://github.com/go-gitea/gitea/assets/38858375/ea5551a7-80ae-4743-bd79-445489a02345)

# Question
So the question is: Is it intended that the statuses doesn't render markdown string?  
If the answer is no I think we could fix it like I did.

I've also looked and did not see any other issue mentioning that.  
Tell me if you need more information :wave:  :cowboy_hat_face:

**Comment by lunny** [comment]
I think it's 255 but without restrict limitations.

**Comment by techknowlogick** [comment]
The field in the status table is meant for a short non-styled message, but we've previously discussed a details page with a more thorough response (see: how github has their "checks" page where 3rd party systems and actions can submit a markdown document for rendering with tables, etc. and can be used for things such as code coverage reports, and such).

**Comment by lunny** [comment]
That should be another column. For this column, we may need to replace \n to blank?

**Comment by alexandreSalconiDenis** [comment]
The tool that send this payload is a github plugin, [github-pr-coverage-status](https://www.jenkins.io/doc/pipeline/steps/github-pr-coverage-status/). If there is a change of endpoint or other discussion can you link the releated PR ?

**Comment by alexandreSalconiDenis** [comment]
> What is currently accepted for that description and how does multiline content render there currently?

@silverwind I'm not sure to understand what you mean by accepted description? and the current result is : the first picture in my pr ? (its a markdown raw)

**Comment by sebastian-sauer** [comment]
Ok - so one possible solution would be to add a checks API (similar to the GitHub API) and add a new tab to display these checks. The checks would behave like a status (so they could be required to merge...).

**Comment by silverwind** [comment]
> Is it intended that the statuses doesn't render markdown string?

I would say yes, the status is meant to be a single line summary string and if you need more, you can link to a external site with more content. That's also how GitHub does it:

https://docs.github.com/en/rest/commits/statuses?apiVersion=2022-11-28#create-a-commit-status

<img width="278" alt="image" src="https://github.com/go-gitea/gitea/assets/115237/6d9fa2db-8a66-4b0a-8c0b-94dd36d6891e">

I don't think it'd be wise to diverge from GitHub here as we want to stay compatible for various reasons and rendering Markdown where it's not expected is a borderline breaking change imho.

**Comment by sebastian-sauer** [comment]
Totally agree. I think for these information we would need the check entity like github uses.

The jenkins plugin to publish checks just uses the gitea status as there is no checks api for gitea.

**Comment by silverwind** [comment]
What could maybe be acceptable would be adding a new optional field like `body`. `description` should be single-line content only and be rendered as such.

**Comment by alexandreSalconiDenis** [comment]
@silverwind would `description` be compatible with plugin from github? @sebastian-sauer seems to say the current situation is just a fall back since we dont have a check page.

**Comment by laraSalconiDenis** [comment]
For some reason on version 1.25.3 the `RenderMarkdownToHtml` doesnt seems to exist. if anyone know a solution it would be nice 🙏 .

**Comment by silverwind** [comment]
`RenderMarkdownToHtml` was renamed to `ctx.RenderUtils.MarkdownToHtml` in https://github.com/go-gitea/gitea/commit/4a469c8e1b3c1d153316aec9fd8cbdd3e1dd54e8.

**Comment by wxiaoguang** [comment]
The answer to "render status as markdown" is no.

The reported commit status should be a single line simple text.

----

I agree with a new field containing the details which can be rendered as markdown.

**Comment by silverwind** [comment]
Yeah I'd say we have to reject this, also for the sake of keeping this compatible with GitHub which specifies `description` as a "A short description of the status" and our UI is also tailored to render only short strings there. More detail can be provided at `target_url`.

**Comment by silverwind** [comment]
And no, I wouldn't add a `details` either, keeping compatible with GitHub API is more important.

https://docs.github.com/en/rest/commits/statuses?apiVersion=2022-11-28#:~:text=Body%20parameters
